git-tar/function/ops.go

```diff
@@ -274,16 +274,39 @@ func clone(pushEvent sdk.PushEvent) (string, error) {
 		return "", fmt.Errorf("cannot create user-dir: %s", userDir)
 	}
 
-	at := &githubAuthToken{
-		appID:          os.Getenv("github_app_id"),
-		installationID: pushEvent.Installation.ID,
-		privateKeyPath: sdk.GetPrivateKeyPath(),
+	var cloneURL string
+
+	if pushEvent.SCM == "github" {
```

Comment:
This would be a good place to use a switch.

As it exapands it probably also makes sense to move the logic for each case to a separate method

Reply:
Yeah, this is a great Idea I will move it to separate functions